src/lib/flight_tasks/tasks/Orbit/FlightTaskOrbit.cpp

```diff
@@ -224,5 +231,37 @@ void FlightTaskOrbit::generate_circle_setpoints(Vector2f center_to_position)
 	_position_setpoint(0) = _position_setpoint(1) = NAN;
 
 	// yawspeed feed-forward because we know the necessary angular rate
-	_yawspeed_setpoint = _v / _r;
+	_yawspeed_setpoint = NAN; //_v / _r;
+
+	switch (_yaw_behaviour) {
+	case orbit_status_s::ORBIT_YAW_BEHAVIOUR_HOLD_INITIAL_HEADING:
+		// make vehicle keep the same heading as when the orbit was commanded
+		_yaw_setpoint = _initial_heading;
+		break;
+
+	case orbit_status_s::ORBIT_YAW_BEHAVIOUR_UNCONTROLLED:
+		// no yaw setpoint
+		_yaw_setpoint = NAN;
+		break;
+
+	case orbit_status_s::ORBIT_YAW_BEHAVIOUR_HOLD_FRONT_TANGENT_TO_CIRCLE:
+		if (_r > 0) {
+			_yaw_setpoint = atan2f(center_to_position(1), center_to_position(0)) + M_PI_F / 2.f;
+
+		} else {
+			_yaw_setpoint = atan2f(center_to_position(1), center_to_position(0)) - M_PI_F / 2.f;
+		}
+
+		break;
+
+	case orbit_status_s::ORBIT_YAW_BEHAVIOUR_RC_CONTROLLED:
```

Comment:
For the front tangential and front to center yaw behavior you want a yawspeed feed forward which was there before and is now commented out otherwise the tracking performance in real world scenarios significantly degrades.